ras_backstage/authentication/token_decoder.py

```diff
@@ -0,0 +1,35 @@
+import logging
+
+from flask import abort
+import jwt
+
+from ras_backstage.authentication.uaa import get_uaa_public_key
+
+
+logger = logging.getLogger(__name__)
+
+USER_ID_CLAIM = "user_id"
+
+
+def decode_access_token(access_token, verify=True):
+
+    try:
+        decoded_jwt = jwt.decode(
+            access_token,
+            verify=verify,
+            algorithms=None,
+            key=get_uaa_public_key(),
+            audience='ras_backstage',
+            leeway=10,
+        )
+        return decoded_jwt
+    except jwt.DecodeError:
+        logger.error(f"Unable to decode token {access_token} - confirm the UAA public key is correct")
+        abort(500)
+
+
+def get_user_id(access_token, verify=True):
```

Comment:
Actually that's not entirely correct, the unit tests pass False to this method. The reasoning being the hard coded token in the tests has expired and therefore would fail verification.

I can remove the test and that will allow me to remove the default parameter?

Reply:
We shouldn't be adding parameters that are only used by tests. I'm not sure we should be testing the decode of a UAA token (`decode_access_token` should probably be `_decode_access_token` and therefore not tested directly in tests). The test could test that given a decoded UAA token, we can get the user id out?

Reply:
I actually disagree on that, its not an uncommon thing to write code in such a way to make it unit testable, I used to do it with protected methods in Java all the time. 

However I have changed it, remove the flag, simplified the code using James original suggestion of removing the get_user_id method and doing a key look up on the returned dict instead. At that point the tests are redundant anyway as they were only testing PyJwt decode method then, so I've removed them.

Reply:
Ok, we should discuss this as a team next week.